core/terminal.go

```diff
@@ -39,7 +39,19 @@ func (term *Terminal) WebsocketURL() string {
 	return fmt.Sprintf("ws://dagger/%s", term.Endpoint)
 }
 
-func (container *Container) Terminal(svcID *idproto.ID, args []string) (*Terminal, http.Handler, error) {
+type TerminalArgs struct {
+	Cmd []string `default:"[]"`
+
+	// Provide dagger access to the executed command
+	// Do not use this option unless you trust the command being executed.
+	// The command being executed WILL BE GRANTED FULL ACCESS TO YOUR HOST FILESYSTEM
+	ExperimentalPrivilegedNesting *bool `default:"false"`
+
+	// Grant the process all root capabilities
+	InsecureRootCapabilities *bool `default:"false"`
```

Comment:
Ah fun, I guess these need to be pointers so we can tell if you meant to override back to `false`.

Reply:
Exactly, I was confused at first but then I realised it was the only solution because we're using `boolean` :/